libs/agno/agno/models/openai/responses.py

```diff
@@ -357,32 +358,29 @@ def _format_messages(self, messages: List[Message]) -> List[Dict[str, Any]]:
 
                 formatted_messages.append(message_dict)
 
-            if self.id.startswith(("o3", "o4-mini")):
-                if message.role == "tool":
-                    if message.tool_call_id and message.content is not None:
-                        formatted_messages.append(
-                            {"type": "function_call_output", "call_id": message.tool_call_id, "output": message.content}
-                        )
-
-            else:
-                # OpenAI expects the tool_calls to be None if empty, not an empty list
-                if message.tool_calls is not None and len(message.tool_calls) > 0:
-                    for tool_call in message.tool_calls:
-                        formatted_messages.append(
-                            {
-                                "type": "function_call",
-                                "id": tool_call["id"],
-                                "call_id": tool_call["call_id"],
-                                "name": tool_call["function"]["name"],
-                                "arguments": tool_call["function"]["arguments"],
-                                "status": "completed",
-                            }
-                        )
-
-                if message.role == "tool":
+            elif message.role == "tool":
+                if message.tool_call_id and message.content is not None:
                     formatted_messages.append(
                         {"type": "function_call_output", "call_id": message.tool_call_id, "output": message.content}
                     )
+            elif message.tool_calls is not None and len(message.tool_calls) > 0:
+                for tool_call in message.tool_calls:
+                    formatted_messages.append(
+                        {
+                            "type": "function_call",
+                            "id": tool_call["id"],
+                            "call_id": tool_call["call_id"],
+                            "name": tool_call["function"]["name"],
+                            "arguments": tool_call["function"]["arguments"],
```

Comment:
Should we handle potential `None`s here?

Reply:
We don't today... if we don't have that info it would fail regardless?